### ISSUE TITLE
Util make problem

### ISSUE THREAD

**Issue by kraymy** [sammy007/open-ethereum-pool#463]
util/util.go:38:16: undefined: common.ToHex
Makefile:10: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by blockinator** [comment]
I am getting the same issue.

**Comment by gudmvik** [comment]
constantly ethash-mining-pool/util  util/util.go:38:16: undefined: common.ToHex

**Comment by Simkas45** [comment]
tell me how to solve this problem?

**Comment by elbochi22** [comment]
same issue, please help!! tnx

**Comment by gudmvik** [comment]
hi. started a pool in the mannet network, go version 15.2, node V. 12. x,core-geth Kepkeriwat (v1.11.17) , lolminer,  (./geth --classic --nat extip:**** --port 30303 --http.port 8545 --http --http.corsdomain "*" --http.addr "192.168.99.100" --maxpeers 300 --syncmode "fast" --http.api "admin,eth,miner,web3,personal" --miner.etherbase "***" --cache=8192 --mine) . did not see any errors, worked 12 hours, CurrentRound Variation: 130%, max share 4 => 50T, 143T , 75T, 64T . No new blocks yet

**Comment by gudmvik** [comment]
Yes, the pool installed yesterday, after the last update : https://github.com/etclabscore/open-etc-pool

**Comment by gudmvik** [comment]
Sammy Thank you for correcting the classic pool , the first block is accepted . America and Russia friendship to hell with politics !

**Comment by bitcoinfever** [comment]
Hi, still have the problem ethash-mining-pool/util util/util.go:38:16: undefined: common.ToHex with open-ethereum-pool ( not the ETC pool )

**Comment by pdieppa** [comment]
I'm also having the same issue while running make under open-ethereum-pool directory.
Fresh install (3 times now) thinking I'm missing something, but always end up with the same error.  Help please.

**Comment by MiningCryptoLive** [comment]
> Hi, still have the problem ethash-mining-pool/util util/util.go:38:16: undefined: common.ToHex with open-ethereum-pool ( not the ETC pool )

If you are trying to make open-ethereum-pool you will need to make changes in util.go and rpc.go to build it.
in rpc.go you need to change line 15 from "github.com/ethereum/go-ethereum/common" TO "github.com/ethereum/go-ethereum/common/hexutil"
Then on line 180 change 
rpcResp, err := r.doPost(r.Url, "eth_sign", []string{from, common.ToHex(hash[:])}) 
TO:
rpcResp, err := r.doPost(r.Url, "eth_sign", []string{from, hexutil.Encode(hash[:])})

In util.go line 10 add: "github.com/ethereum/go-ethereum/common/hexutil"

line 38 change 
return string(common.ToHex(diff1.Bytes()))
 TO:
 return string(hexutil.Encode(diff1.Bytes()))

then make clean and make and you should be good to go

**Comment by bitcoinfever** [comment]
Thank you, now I am getting another error:
Makefile:10: die Regel für Ziel „all“ scheiterte
make: *** [all] Fehler 1

**Comment by honorkeles** [comment]
Hi @Racing1 

I get below error when I made the changes that you mentioned. Please advice. 

I would appreciate for any help

build/env.sh go get -v ./...
github.com/sammy007/open-ethereum-pool/util
# github.com/sammy007/open-ethereum-pool/util
util/util.go:10:2: imported and not used: "github.com/ethereum/go-ethereum/common/hexutil"
util/util.go:38:16: undefined: common.ToHex
Makefile:10: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by Racing1** [comment]
I just did a clean build with the changes and it works fine. Make sure to make clean after making the changes to the util.go and rpc.go files

**Comment by MiningCryptoLive** [comment]
> Hi @Racing1
> 
> I get below error when I made the changes that you mentioned. Please advice.
> 
> I would appreciate for any help
> 
> build/env.sh go get -v ./...
> github.com/sammy007/open-ethereum-pool/util
> 
> # github.com/sammy007/open-ethereum-pool/util
> util/util.go:10:2: imported and not used: "github.com/ethereum/go-ethereum/common/hexutil"
> util/util.go:38:16: undefined: common.ToHex
> Makefile:10: recipe for target 'all' failed
> make: *** [all] Error 2

I just did a clean build with the changes and it works fine. Make sure to make clean after making the changes to the util.go and rpc.go files

**Comment by honorkeles** [comment]
when you say clean build, what did you do? 

I just entered 'make clean' and it returns below. Is there more steps to have a clean build?

"rm -fr build/_workspace/pkg/ build/bin/*"

**Comment by MiningCryptoLive** [comment]
Make clean.

**Comment by MiningCryptoLive** [comment]
Yes, Make Clean

**Comment by MiningCryptoLive** [comment]
Then Make

**Comment by honorkeles** [comment]
yes, I did it but getting the error still. 

can you share util.go ?

**Comment by MiningCryptoLive** [comment]
When I get home to a computer I can

**Comment by MiningCryptoLive** [comment]
Is the rpc.go ok

**Comment by honorkeles** [comment]
yeah rpc.go no error and I found my mistake on util.go. There is no error after "make" but there is no config.json in the folder. Any clue?


after running "make" , only runs below and goes back to folder.
build/env.sh go get -v ./...

**Comment by MiningCryptoLive** [comment]
> yeah rpc.go no error and I found my mistake on util.go. There is no error after "make" but there is no config.json in the folder. Any clue?
> 
> after running "make" , only runs below and goes back to folder.
> build/env.sh go get -v ./...

should be a config.example.json file you can rename config.json

**Comment by honorkeles** [comment]
thanks man. I did it.

I get below errors when I run the pool. I added 8545 with ufw allow 8545/tcp but still get connection refused

Error while refreshing pending block on main: Post "http://127.0.0.1:8545": dial tcp 127.0.0.1:8545: connect: connection refused
Error while refreshing pending block on main: Post "http://127.0.0.1:8545": dial tcp 127.0.0.1:8545: connect: connection refused
Error while refreshing pending block on main: Post "http://127.0.0.1:8545": dial tcp 127.0.0.1:8545: connect: connection refused

**Comment by honorkeles** [comment]
I figured it out. I needed to run     Geth --rpc --mine

thanks @Racing1

**Comment by MiningCryptoLive** [comment]
> I figured it out. I needed to run Geth --rpc --mine
> 
> thanks @Racing1

No problem. Glad you got it sorted out

**Comment by MiningCryptoLive** [comment]
whats your pool ip so I can check it out?

**Comment by crypto623** [comment]
If error is "go: cannot find main module;", please refer [here](https://github.com/golang/go/issues/31997#issuecomment-782864390)

**Comment by fuyundian** [comment]
> Yes, Make Clean
I've deleted it a couple of times, but there's also undefined: common.tohex
